routing/road_access.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace routing
+{
+// This class provides information about road access classes.
+// For now, only restrictive types (such as barrier=gate and access=private)
+// and only car routing are supported.
+class RoadAccess final
+{
+public:
+  std::vector<uint32_t> & GetPrivateRoads() { return m_privateRoads; }
+  std::vector<uint32_t> const & GetPrivateRoads() const { return m_privateRoads; }
+
+  void StealPrivateRoads(std::vector<uint32_t> && v) { m_privateRoads = std::move(v); }
```

Comment:
It's not Steal. It's Set. And I strongly suggest you to use rvalue reference with perfect forwarding.